src/components/EventList.tsx

```diff
@@ -97,30 +114,87 @@ const EventListItem = ({ event, response, onSignup, onUndoSignup, org }: EventLi
                 />
             </View>
             <View data-testid="location-title">{ event.location.title }</View>
-            { user ? response ? (
+            { user ? (
+                <EventResponseButton
+                    booked={ booked }
+                    event={ event }
+                    onSignup={ onSignup }
+                    onUndoSignup={ onUndoSignup }
+                    org={ org }
+                    response={ response }
+                    todo={ todo }
+                />
+            ) : <SignupDialogTrigger /> }
+            <NextLink href={ `/o/${org.id}/events/${ event.id }` }>
+                <a>
+                    <Button marginTop="size-50" variant="cta">
+                        <Msg id="misc.eventList.moreInfo" />
+                    </Button>
+                </a>
+            </NextLink>
+        </Flex>
+    );
+};
+
+interface EventResponseButtonProps {
+    booked: BookedEvent | undefined;
+    event: ZetkinEvent;
+    onSignup: (eventId: number, orgId: number) => void;
+    onUndoSignup: (eventId: number, orgId: number) => void;
+    org?: ZetkinOrganization;
```

Comment:
The `org` property is specified as optional here, but is then actually required (and asserted to not be null) in several places within the component. It should not be flagged as optional here.